src/lib/cimb_translator/CimbReader.cpp

```diff
@@ -43,20 +45,66 @@ namespace {
 		bitmatrix::mat_to_bitbuffer(symbols, bb.get_writer());
 		return bb;
 	}
+
+	void updateMaxColor(std::tuple<float, float, float>& max_color, const cv::Scalar& c)
+	{
+		std::get<0>(max_color) = std::max(std::get<0>(max_color), static_cast<float>(c[0]));
+		std::get<1>(max_color) = std::max(std::get<1>(max_color), static_cast<float>(c[1]));
+		std::get<2>(max_color) = std::max(std::get<2>(max_color), static_cast<float>(c[2]));
+	}
+
+	std::tuple<float, float, float> calculateWhite(const cv::Mat& img, bool dark)
+	{
+		std::tuple<float, float, float> bestColor({1, 1, 1});
+		if (dark)
+		{
+			unsigned tl = Config::anchor_size() - 2;
+			unsigned br = Config::image_size() - Config::anchor_size() - 2;
+			std::array<std::pair<unsigned, unsigned>, 3> anchors = {{ {tl, tl}, {tl, br}, {br, tl} }};
```

Comment:
Use the 3 anchor corners of the image to find a reasonable "observed" white value. The anchor pattern should have a white patch in the middle -- if we exacted it properly, it'll be at these locations. We'll use the maximum RGB values we find.